clients/src/main/java/org/apache/kafka/common/config/provider/DirectoryConfigProvider.java

```diff
@@ -44,8 +44,15 @@ public class DirectoryConfigProvider implements ConfigProvider {
 
     private static final Logger log = LoggerFactory.getLogger(DirectoryConfigProvider.class);
 
+    public static final String ALLOWED_PATHS_CONFIG = "allowed.paths";
+    public static final String ALLOWED_PATHS_DOC = "A comma separated list of paths that this config provider is " +
+            "allowed to access. If not set, all paths are allowed.";
+    private AllowedPaths allowedPaths = new AllowedPaths(null);
```

Comment:
This class is required to be thread-safe in the ConfigProvider javadoc.

```suggestion
    private volatile AllowedPaths allowedPaths = new AllowedPaths(null);
```